doc/sphinxext/missing_references.py

```diff
@@ -97,7 +97,10 @@ def get_location(node, app):
     (path, line) = get_source_line(node)
 
     if path:
-
+        # sometimes the 'path' can contain ':' which are forbidden on
+        # windows, but on posix just passes through.
+        path, *post = path.partition(':')
```

Comment:
uh, how can there be colons in the path, even on posix?  a priori this should be a real filesystem path given that we call Path(path).resolve() just below

Reply:
Yes, it's confusing that this is named "path", while in reality just being some string which *should* contain a path at the start.

Reply:
As always I'm a bit confused by the magic of escaping, but `:` appears to be kosher in posix paths:

```
sharon@21:09  ➤  touch "/tmp/path:o:log:i:cal"
sharon@09:03  ➤  echo "hi there" >  "/tmp/path:o:log:i:cal"
sharon@09:03  ➤  ls /tmp/ | grep path
path:o:log:i:cal
sharon@09:03  ➤  cat /tmp/path\:o\:log\:i\:cal 
hi there

```

Reply:
It's certainly kosher (anything but / and \0 are for posix), but how is that relevant here? if we're running on linux, then the path is what it is and it will become relative to the mpl root just below (in relative_to), and there are no directories in mpl with colons; if running on windows, get_source_line should be returning a real path on the windows filesystem and that has no colons.